src/subgraph/zns/actions/getSubdomainsById.ts

```diff
@@ -34,12 +34,16 @@ export const getSubdomainsById = async <T>(
     );
 
     queriedDomains = queryResult.data.domains;
-    if (queriedDomains.length === 0) {
-      return [] as Domain[];
-    }
+    
     for (const domain of queriedDomains) {
       domains.push(convertDomainDtoToDomain(domain));
     }
+
+    // If there were no new queried domains, we can stop querying
+    if (queriedDomains.length === 0) {
+      return domains
```

Comment:
only issue i can see is that by returning early, we're missing out on the log message. not sure how often this comes up, but i think it's worth ensuring the logging is consistent to avoid confusion.